wrapper/src/main/java/software/amazon/jdbc/hostlistprovider/RdsHostListProvider.java

```diff
@@ -278,66 +258,36 @@ public FetchTopologyResult getTopology(final Connection conn, final boolean forc
     }
   }
 
-  protected void clusterIdChanged(final String oldClusterId) {
-    // do nothing
-  }
+  protected String getMatchedClusterId(final @NonNull List<HostSpec> hosts) {
+    if (Utils.isNullOrEmpty(hosts)) {
+      return null;
+    }
 
-  protected ClusterSuggestedResult getSuggestedClusterId(final String url) {
-    for (final Entry<String, List<HostSpec>> entry : topologyCache.getEntries().entrySet()) {
-      final String key = entry.getKey(); // clusterId
-      final List<HostSpec> hosts = entry.getValue();
-      final boolean isPrimaryCluster = primaryClusterIdCache.get(key, false,
-          suggestedClusterIdRefreshRateNano);
-      if (key.equals(url)) {
-        return new ClusterSuggestedResult(url, isPrimaryCluster);
+    for (HostSpec hostSpec : hosts) {
+      // TODO: use template pattern port?
```

Comment:
Is this going to be investigated/addressed later or can this comment be removed?